### PR TITLE
chore: lint consistent type imports

### PR DESCRIPTION
**Pull request by rarkins**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
@viceice do you think this is worth adding? Causes ~150 errors in our current code which will need fixing

**Comment by rarkins** [comment]
Now pushed automated fixes + manual ignores to get eslint passing

**Review by viceice — DISMISSED** [review]
what's the benefit? 🤔

File: lib/modules/platform/codecommit/index.spec.ts [hunk 1]
```diff
@@ -24,7 +24,8 @@ import {
 import * as git from '../../../util/git';
 import type { Platform } from '../types';
 import { getCodeCommitUrl } from './codecommit-client';
-import { CodeCommitPr, config } from './index';
+import type { CodeCommitPr } from './index';
+import { config } from './index';
```
Comment (viceice) [inline lib/modules/platform/codecommit/index.spec.ts:28]:
```suggestion
import type { CodeCommitPr } from '.';
import { config } from '.';
```

File: lib/modules/platform/gitea/index.spec.ts [hunk 2]
```diff
@@ -42,10 +42,10 @@ const GITEA_VERSION = '1.14.0+dev-754-g5d2b7ba63';
 
 describe('modules/platform/gitea/index', () => {
   let gitea: Platform;
-  let helper: jest.Mocked<typeof import('./gitea-helper')>;
+  let helper: jest.Mocked<typeof import('./gitea-helper')>; // eslint-disable-line @typescript-eslint/consistent-type-imports
```
Comment (viceice) [inline lib/modules/platform/gitea/index.spec.ts:45]: can this be configured to ignore dynamic imports?

File: lib/modules/versioning/ruby/version.ts [hunk 3]
```diff
@@ -1,8 +1,6 @@
 import { eq, major, minor, patch, prerelease } from '@renovatebot/ruby-semver';
-import {
-  SegmentElement,
-  create,
-} from '@renovatebot/ruby-semver/dist/ruby/version.js';
+import type { SegmentElement } from '@renovatebot/ruby-semver/dist/ruby/version.js';
+import { create } from '@renovatebot/ruby-semver/dist/ruby/version.js';
```
Comment (viceice) [inline lib/modules/versioning/ruby/version.ts:3]: can you open an issue on the ruby semver repo to update exports? so we don't need those deep imports?

File: lib/workers/repository/dependency-dashboard.spec.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { ERROR, WARN } from 'bunyan';
 import { mock } from 'jest-mock-extended';
 import { Fixtures } from '../../../test/fixtures';
+import type { RenovateConfig } from '../../../test/util';
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.spec.ts:4]: we probably should remove the export from test util and use the original source

**Comment by rarkins** [comment]
@viceice I thought you'd like this one, but happy to drop it if it's not worth the hassle

**Pull request closed without merging**